signer/src/stacks/contracts.rs

```diff
@@ -987,11 +1070,118 @@ impl AsContractCall for RejectWithdrawalV1 {
+        //    bitcoin_withdrawals_outputs table, and that any of the associated txids are
+        //    confirmed on the canonical bitcoin blockchain. Fail the withdrawal request if
+        //    such a transaction was found.
+
+        let stacks_chain_tip = ctx
+            .get_storage()
+            .get_stacks_chain_tip(&req_ctx.chain_tip.block_hash)
+            .await
+            .unwrap()
+            .unwrap();
+        let Some(report) = ctx
+            .get_storage()
+            .get_withdrawal_request_report(
+                &req_ctx.chain_tip.block_hash,
+                &stacks_chain_tip.block_hash,
+                &self.id,
+                &ctx.config().signer.public_key(),
+            )
+            .await
+            .map_err(|_| WithdrawalRejectErrorMsg::RequestMissing.into_error(req_ctx, self))?
+        else {
+            return Err(WithdrawalRejectErrorMsg::RequestMissing.into_error(req_ctx, self));
+        };
+
+        match report.status {
+            WithdrawalRequestStatus::Confirmed => (),
+            WithdrawalRequestStatus::Fulfilled(_txid) => {
+                // fails #2
+                return Err(
+                    WithdrawalRejectErrorMsg::WithdrawalRequestFulfilled.into_error(req_ctx, self)
+                );
+            }
+            WithdrawalRequestStatus::Unconfirmed => {
+                // fails #1
+                return Err(WithdrawalRejectErrorMsg::WithdrawalRequestUnconfirmed
+                    .into_error(req_ctx, self));
+            }
+        }
+
+        // 3. The request is final. Check that we've seen six new bitcoin blocks since observing
+        //    the bitcoin anchor block associated with the Stacks block confirming the withdrawal
+        //    request transaction. Fail the withdrawal request if we’ve observed less than 6
+        //    bitcoin blocks since the anchor block.
+
+        let request_block_height = report.block_height;
+
+        let Some(tip_block_hash) = ctx.get_storage().get_bitcoin_canonical_chain_tip().await?
+        else {
+            return Err(WithdrawalRejectErrorMsg::NoBitcoinChainTip.into_error(req_ctx, self));
+        };
+
+        let Some(tip_bitcoin_block) = ctx.get_storage().get_bitcoin_block(&tip_block_hash).await?
+        else {
+            return Err(WithdrawalRejectErrorMsg::MissingBitcoinBlock.into_error(req_ctx, self));
+        };
+
+        let tip_block_height = tip_bitcoin_block.block_height;
+
+        if tip_block_height < (request_block_height + 6) {
+            return Err(
+                WithdrawalRejectErrorMsg::WithdrawalRequestNotFinal.into_error(req_ctx, self)
+            );
+        }
+
+        // 4. The request is collectively rejected or expired.
+        let signer_votes = ctx
+            .get_storage()
+            .get_withdrawal_request_signer_votes(&self.id, &req_ctx.aggregate_key)
+            .await
+            .map_err(|_| WithdrawalRejectErrorMsg::RequestMissing.into_error(req_ctx, self))?;
```

Comment:
You should just bail here.
```suggestion
            .await?;
```